src/main/java/io/stargate/sgv2/jsonapi/service/operation/OperationAttempt.java

```diff
@@ -634,6 +613,24 @@ public PrettyToStringBuilder toString(PrettyToStringBuilder prettyToStringBuilde
         .append("failure", failure);
   }
 
+  /**
+   * A statement that a task is going to run, and a supplier that will run it when called.
+   *
+   * <p>This is really just a Pair so we can give the Statement used to the {@link
+   * DefaultDriverExceptionHandler} via the {@link DefaultDriverExceptionHandler.Factory} and the
+   * statement context into the exception handling.
+   *
+   * @param statement Statement used with the attempt, may be null.
+   * @param results A supplier that fetches the results when queried, the supplier must wait until
+   *     called to avoid blocking. Must not be null.
+   */
+  protected record StatementContext(
```

Comment:
This is pretty well-hidden here (took a while to find via GH diffs) -- while it is sort of internal to `OperationAttempt` should it be a main-level record or (alternatively) at beginning of this class next to `OperationStatus`? Our codebase is pretty heavy on inner classes/records.
(but I can see it might make sense for non-public types).

Reply:
agree it can be better, and am not happy with the name etc. 

Am working on some refactoring for this areas so we can handle the embedding in the pipeline better and will improve as part of that.